src/system/api/rqidApi.ts

```diff
@@ -114,12 +116,44 @@ export class Rqid {
       result.splice(0, 0, ...worldDocuments);
     }
 
+    // const distinctWorldRqids: string[] = result
+    //   .map((d) => d.data?.flags?.rqg?.documentRqidFlags?.id)
+    //   .filter(function (value, index, self) {
+    //     return self.indexOf(value) === index;
+    //   });
+
+    const distinctWorldRqids: string[] = [
+      ...new Set(result.map((d) => d.data?.flags?.rqg?.documentRqidFlags?.id)),
+    ];
+
+    // find the best results in the world
+    if (onlyFindBest) {
+      result = [];
+      for (const rqid of distinctWorldRqids) {
+        const bestWorldDocFound = await Rqid.fromRqid(rqid);
+        if (bestWorldDocFound) {
+          result.push(bestWorldDocFound);
+        }
+      }
+    }
+
     if (["match", "all", "compendiums"].includes(scope)) {
-      const compendiaDocuments = await Rqid.documentsFromCompendia(
-        rqidRegex,
-        rqidDocumentType,
-        lang
-      );
+      let compendiaDocuments = await Rqid.documentsFromCompendia(rqidRegex, rqidDocumentType, lang);
+
+      if (onlyFindBest) {
+        // list of all the rqids from the compendia that were not already found in the world.
+        const distinctCompendiaRqids: string[] = [
+          ...new Set(compendiaDocuments.map((d) => d.data?.flags?.rqg?.documentRqidFlags?.id)),
+        ];
+        compendiaDocuments = [];
+        for (const rqid of distinctCompendiaRqids) {
+          const bestCompendiaDocFound = await Rqid.fromRqid(rqid);
```

Comment:
I don't think you can use fromRqid here, since that will stop looking if there are any world documents. So if you specify you want "all" and there exists a world document, then you won't get any compendium documents I think.
On the other hand you already have the documents in `compendiaDocuments` so there should be no need to fetch them again, `documentsFromCompendia` should already have handled that.

Reply:
@wake42 I forgot to push to the PR. Please look again.  But I DO want it to stop trying to find documents and not look in compendium packs if it finds a document in the World, because documents in the World always beat ones in the packs.